src/System.Management.Automation/engine/Utils.cs

```diff
@@ -990,7 +990,32 @@ internal static bool NativeItemExists(string path, out bool isDirectory, out Exc
             if (result == -1)
             {
                 int errorCode = Marshal.GetLastWin32Error();
-                if (errorCode == 5)
+
+                // Errorcode 32 is 'ERROR_SHARING_VIOLATION' i.e. 
+                // The process cannot access the file because it is being used by another process.
+                // GetFileAttributes will return INVALID_FILE_ATTRIBUTES for a system file or directory because of this error.
+                // Fallback to FindFirstFile to check if the file actually exists or not.
+                if (errorCode == 32)
+                {
+                    IntPtr INVALID_HANDLE_VALUE = new IntPtr(-1);
+                    NativeMethods.WIN32_FIND_DATA findData;
+                    IntPtr findHandle;
+                    findHandle = NativeMethods.FindFirstFile(path, out findData);
+                    NativeMethods.FindClose(findHandle);
```

Comment:
Although FindClose() internally takes care if the handle is invalid. But in our case , we are anyways checking if the handle is valid, so calling FindClose() only in case it's valid.